Sources/Plasma/CoreLib/HeadSpin_Mac.mm

```diff
@@ -101,39 +104,19 @@ int hsMessageBoxWithOwner(hsWindowHndl owner, NSString* message, NSString* capti
         [lock wait];
         [lock unlock];
     }
-    
+
     return (int)response;
 }
 
-int hsMessageBoxWithOwner(hsWindowHndl owner, const char* message, const char* caption, int kind, int icon)
+int hsMessageBoxWithOwner(hsWindowHndl owner, const ST::string& message, const ST::string& caption, int kind, int icon)
 {
     if (hsMessageBox_SuppressPrompts)
         return hsMBoxOk;
-    
-    @autoreleasepool {
-        return hsMessageBoxWithOwner(owner,
-                                     [NSString stringWithCString:caption encoding:NSUTF8StringEncoding],
-                                     [NSString stringWithCString:message encoding:NSUTF8StringEncoding],
-                                     kind,
-                                     icon);
-    }
-}
 
-int hsMessageBoxWithOwner(hsWindowHndl owner, const wchar_t* message, const wchar_t* caption, int kind, int icon)
-{
-    if (hsMessageBox_SuppressPrompts)
-        return hsMBoxOk;
-    
     @autoreleasepool {
         return hsMessageBoxWithOwner(owner,
-                                     [[[NSString alloc] initWithBytesNoCopy:(void *)caption
-                                                                     length:wcslen(caption)*sizeof(*caption)
-                                                                   encoding:NSUTF32LittleEndianStringEncoding freeWhenDone:NO
-                                      ] autorelease],
-                                     [[[NSString alloc] initWithBytesNoCopy:(void *)message
-                                                                     length:wcslen(caption)*sizeof(*caption)
-                                                                   encoding:NSUTF32LittleEndianStringEncoding freeWhenDone:NO
-                                      ] autorelease],
+                                     [NSString initWithUTF8String:message.c_str()],
+                                     [NSString initWithUTF8String:caption.c_str()],
```

Comment:
```suggestion
                                     [NSString stringWithSTString:message],
                                     [NSString stringWithSTString:caption],
```

Reply:
We can't use that here (reliably) because NSString+STString category is in plClient, and this might (hypothetically) get called from somewhere that's not part of plClient.

Sure would be convenient if we could do something like `message.toNSString()` as part of the base library :trollface: